enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingService.java

```diff
@@ -316,6 +318,58 @@ public void terminateInactiveProcesses() {
         otpService.terminateOldOtps(createdDateOtp);
     }
 
+    /**
+     * Provide consent text.
+     *
+     * @param request consent text request
+     * @return consent response
+     */
+    public OnboardingConsentTextResponse fetchConsentText(final OnboardingConsentTextRequest request) throws OnboardingProcessException {
+        final ConsentTextRequest providerRequest = ConsentTextRequest.builder()
+                .processId(request.getProcessId())
+                .userId(request.getUserId())
+                .consentType(convert(request.getConsentType()))
+                .locale(LocaleUtils.toLocale(request.getLanguage()))
+                .build();
+
+        try {
+            final String consentText = onboardingProvider.fetchConsent(providerRequest);
+            final OnboardingConsentTextResponse response = new OnboardingConsentTextResponse();
+            response.setConsentText(consentText);
+            return response;
+        } catch (OnboardingProviderException e) {
+            throw new OnboardingProcessException("An error when fetching consent text.", e);
+        }
+    }
+
+    /**
+     * Record dis/approval of consent
+     *
+     * @param request approval request
+     */
+    public void approveConsent(final OnboardingConsentApprovalRequest request) throws OnboardingProcessException {
```

Comment:
Let's consider if we can live with void method here.

Reply:
👍  You are right. From compatibility point of view, it's better to return even empty object expandable in the future.

Reply:
Agree with empty return type.